### PR TITLE
Feature/db

### PR DESCRIPTION
**Pull request by Panda-Sharp**
- improved db schema, now users and projects has a N to N relationship
- implemented POST user and POST project

### CODE REVIEW

File: src/api/projects/post.ts [hunk 1]
```diff
@@ -26,20 +26,15 @@ module.exports = (req: Request, res: Response) => {
...
-        
-        let discordId = (user as IDiscordUser).id;
+        let discordId = await GetDiscordToken(req,res);        
 
         submitProject(body, discordId)
-            .then(results => {
-                res.end("Success");
+            .then(() => {
+                res.status(200);
+                res.json(JSON.stringify({
```
Comment (jmurth1234) [inline src/api/projects/post.ts]: So although this _works_, you don't need to `JSON.stringify` an object when using `res.json` as it's done as part of the jsonification. I'll be opening a PR at some point to fix this across the project

File: src/index.ts [hunk 2]
```diff
@@ -25,11 +25,11 @@ const helpers = require('./common/helpers');
 const swaggerUi = require('swagger-ui-express');
 
 const PORT = process.env.PORT || 5000;
-const DEBUG = process.argv.filter(val => val == 'dev').length > 0;
 const MOCK = process.argv.filter(val => val == 'mock').length > 0;
 
 app.use(bodyParser.urlencoded({ extended: true }));
 
+app.use(bodyParser.urlencoded({ extended: true }));
```
Comment (jmurth1234) [inline src/index.ts]: Do we need this twice lol

**Review by jmurth1234 — APPROVED** [review]
Other than the two minor issues commented this is good to go.

As a side note: deleting the `package-lock.json` is generally a bad idea as it ensures all developers and the server always have the exact same packages.

The important bits all look good though 👍

File: src/api/projects/post.ts (hunk 1 above)
Comment (Panda-Sharp) [reply src/api/projects/post.ts]: good catch, it was from some previous code :)

File: src/index.ts (hunk 2 above)
Comment (Panda-Sharp) [reply src/index.ts]: good catch again, probably some merging effect :)

File: src/common/helpers.ts [hunk 3]
```diff
@@ -150,9 +150,24 @@ export async function GetDiscordUser(accessToken: string): Promise<IDiscordUser
     return await Req.json();
 }
 
+export async function GetDiscordToken(req: Request, res: Response): Promise<string> {
```
Comment (Arlodotexe) [inline src/common/helpers.ts]: I suggest renaming this from `GetDiscordToken` to `GetDiscordIdFromToken`

**Review by Arlodotexe — CHANGES_REQUESTED** [review]
Just a quick change, otherwise it looks good!

**Review by Arlodotexe — APPROVED** [review]

**Pull request merged**